File: console/src/access/capabilities.ts
export const Capability = {
  PlatformUse: "platform.use",
  UsersManage: "users.manage",
  PlatformSettingsManage: "platform.settings.manage",
  PublicationsReview: "publications.review",
} as const;

export type Capability = (typeof Capability)[keyof typeof Capability];

export type PlatformRole = "admin" | "member";
export type PlatformMode = "legacy" | "multi_user";

export function can(
  mode: PlatformMode,
  role: PlatformRole | null,
  capability: Capability | string,
): boolean {
  if (mode === "legacy") return true;
  if (capability === Capability.PlatformUse) {
    return role === "admin" || role === "member";
  }
  if (
    capability === Capability.UsersManage ||
    capability === Capability.PlatformSettingsManage ||
    capability === Capability.PublicationsReview
  ) {
    return role === "admin";
  }
  return false;
}

File: src/qwenpaw/access/capabilities.py
# -*- coding: utf-8 -*-
"""平台级 capability 的最小稳定集合。"""

from enum import StrEnum


class Capability(StrEnum):
    """平台与 Agent 入口使用的稳定能力集合。"""

    PLATFORM_USE = "platform.use"
    AGENT_USE = "agent.use"
    USERS_MANAGE = "users.manage"
    PLATFORM_SETTINGS_MANAGE = "platform.settings.manage"
    MODELS_MANAGE = "models.manage"
    SKILLS_MANAGE = "skills.manage"
    PLUGINS_MANAGE = "plugins.manage"
    PUBLICATIONS_REVIEW = "publications.review"
